controllers/rabbitmq/rabbitmq_controller.go

```diff
@@ -356,6 +362,31 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 
 	if clusterReady {
 		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+		// Let's wait DeploymentReadyCondition=True to apply the policy
+		queue := instance.Spec.QueueType
+		ismirrored := instance.Status.QueueType
+
+		if queue == "Mirrored" && *instance.Spec.Replicas > 1 {
+			if ismirrored != "Mirrored" {
+				Log.Info("ha-all policy not present. Applying.")
+				err := updateMirroredPolicy(ctx, helper, instance, r.config, true)
+				if err != nil {
+					Log.Error(err, "Could not apply policy")
+				} else {
+					instance.Status.QueueType = "Mirrored"
+				}
+			}
+		} else {
+			if ismirrored == "Mirrored" {
+				Log.Info("Removing ha-all policy")
+				err := updateMirroredPolicy(ctx, helper, instance, r.config, false)
+				if err != nil {
+					Log.Error(err, "Could not remove policy")
+				} else {
```

Comment:
same the else is not required, or is it. could just set instance.Status.QueueType = "" if no error occured